extensions/AzureCosmosDBMongoVCore/AzureCosmosDBMongoVCore/AzureCosmosDBMongoVCoreConfig.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.KernelMemory;
+
+public class AzureCosmosDBMongoVCoreConfig
+{
+    /// <summary>
+    /// Get more details about Azure Cosmos Mongo vCore https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search
+    /// </summary>
+
+    /// <summary>
+    /// Connection string required to connect to Azure Cosmos Mongo vCore
+    /// </summary>
+    public string ConnectionString { get; set; }
+
+    /// <summary>
+    /// Indexn name for the Mongo vCore DB
+    /// </summary>
+    public string IndexName { get; set; }
+
+    /// <summary>
+    /// Kind: Type of vector index to create.
+    ///     Possible options are:
+    ///         - vector-ivf
+    ///         - vector-hnsw: available as a preview feature only,
+    ///                        to enable visit https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/preview-features
+    /// </summary>
+    public string Kind { get; set; }
+
+    /// <summary>
+    /// NumLists: This integer is the number of clusters that the inverted file (IVF) index uses to group the vector data.
+    /// We recommend that numLists is set to documentCount/1000 for up to 1 million documents and to sqrt(documentCount)
+    /// for more than 1 million documents. Using a numLists value of 1 is akin to performing brute-force search, which has 
+    /// limited performance.
+    /// </summary>
+    public int NumLists { get; set; }
+
+    /// <summary>
+    /// Similarity: Similarity metric to use with the IVF index.
+    ///     Possible options are:
+    ///         - CosmosDBSimilarityType.COS (cosine distance),
+    ///         - CosmosDBSimilarityType.L2 (Euclidean distance), and
+    ///         - CosmosDBSimilarityType.IP (inner product).
+    /// </summary>
+    public string Similarity { get; set; }
+
+    /// <summary>
+    /// Dimensions: Number of dimensions for vector similarity. The maximum number of supported dimensions is 2000
+    /// </summary>
+    public int Dimensions { get; set; }
+
+    /// <summary>
+    /// NumberOfConnections: The max number of connections per layer (16 by default, minimum value is 2, maximum value is 
+    /// 100). Higher m is suitable for datasets with high dimensionality and/or high accuracy requirements.
+    /// </summary>
+    public int NumberOfConnections { get; set; }
```

Comment:
if higher is better, why would one choose 2 instead of 100?

Reply:
If the dimensionality of the dataset is low, then you won't need a higher value of "m(NumberOfConnections)". 

Smaller values of "m" tend to be better for lower recalls or low-dimensional data while higher values of"m" are suited better for high recalls or high-dimensional data.